### ISSUE TITLE
Replace  cublasSgemm with cublasSgemmStridedBatched() to speedup training

### ISSUE THREAD

**Issue by TaihuLight** [AlexeyAB/darknet#1194]

**Comment by AlexeyAB** [comment]
Yes, `cublasSgemmStridedBatched()` is faster than `cublasSgemmBatched()` that is faster than `cublasSgemm()`.

* But `cublasSgemmStridedBatched()` faster than `cublasSgemmBatched()` only when `stride>1` that is only in 5 conv-layer in the Yolo v3, is no in Yolo v2.

* `cublasSgemmBatched()` is faster than `cublasSgemm()` only when `batch>1` i.e. only for Training.

But if we use cuDNN, then this is faster than `cublasSgemmStridedBatched()`.

**Comment by TaihuLight** [comment]
How to call CLBlastSgemmStridedBatched( ) API to implement batched gemm
https://github.com/CNugteren/CLBlast/issues/313

**Comment by TaihuLight** [comment]
How to call CLBlastSgemmStridedBatched( ) API to implement batched gemm?
https://github.com/CNugteren/CLBlast/issues/313
